test/oli/accounts_test.exs

```diff
@@ -104,6 +104,63 @@ defmodule Oli.AccountsTest do
 
       assert author.preferences.timezone == "America/Los_Angeles"
     end
+
+    test "update author password successfully" do
+      author = insert(:author)
+
+      attrs = %{
+        "current_password" => "password_1",
+        "email" => author.email,
+        "family_name" => author.family_name,
+        "given_name" => author.given_name,
+        "password" => "password_2",
+        "password_confirmation" => "password_2"
+      }
+
+      assert {:ok, _author = %Author{}} = Accounts.update_author(author, attrs)
+    end
+
+    test "update author password fails when new password and password confirmation are different" do
+      author = insert(:author)
+
+      attrs = %{
+        "current_password" => "password_1",
+        "email" => author.email,
+        "family_name" => author.family_name,
+        "given_name" => author.given_name,
+        "password" => "password_22",
+        "password_confirmation" => "password_2"
+      }
+
+      assert {:error, changeset} = Accounts.update_author(author, attrs)
+      assert changeset.valid? == false
+
+      {:password_confirmation, {error_message, [validation: :confirmation]}} =
+        List.first(changeset.errors)
+
+      assert error_message == "does not match confirmation"
+    end
+
+    test "update author password fails when the new password is less than 8 characters long" do
+      author = insert(:author)
+
+      attrs = %{
+        "current_password" => "password_1",
+        "email" => author.email,
+        "family_name" => author.family_name,
+        "given_name" => author.given_name,
+        "password" => "pass",
+        "password_confirmation" => "pass"
+      }
+
+      assert {:error, changeset} = Accounts.update_author(author, attrs)
+      assert changeset.valid? == false
+
+      {:password, {error_message, [count: 8, validation: :length, kind: :min, type: :string]}} =
+        List.first(changeset.errors)
```

Comment:
Because there is only one element in `errors` you could match the list directly. Something like:

```
[{:password_confirmation, {error_message, [validation: :confirmation]}}] = changeset.errors
```